src/Viewer.tsx

```diff
@@ -21,30 +22,69 @@ export interface WidgetModelWithState extends WidgetModel {
 
 export interface WidgetProps {
   controller: SimulariumController;
-  height: number;
-  width: number;
 }
 
 function ViewerWidget(props: WidgetProps): JSX.Element {
   const [modelInfo, setModelInfo] = useState<ModelInfo | undefined>({});
   const [trajectoryTitle, setTrajectoryTitle] = useState<string | undefined>(
     ''
   );
+  const [dimensions, setDimensions] = useState({ width: 500, height: 529 });
+  const [showSidePanel, setShowSidePanel] = useState(true);
+
+  const containerRef = useRef<HTMLDivElement>(null);
+  const observerRef = useRef<ResizeObserver | null>(null);
+
+  useEffect(() => {
+    // Initialize ResizeObserver if it doesn't exist
+    if (!observerRef.current && containerRef.current) {
+      observerRef.current = new ResizeObserver(
+        (entries: ResizeObserverEntry[]) => {
+          for (const entry of entries) {
+            // get the size of viewer container
+            let { width } = entry.contentRect;
+            const { height } = entry.contentRect;
+            // hide side panel if space is small
+            setShowSidePanel(width > 580);
+            if (showSidePanel) {
+              width = width - 280;
+            }
+            // pass size to viewer
+            setDimensions({ width, height });
+          }
+        }
+      );
+      // observe the container size
+      observerRef.current.observe(containerRef.current);
+    }
+
+    // Cleanup function
+    return () => {
+      if (observerRef.current) {
+        if (containerRef.current) {
+          observerRef.current.unobserve(containerRef.current);
+        }
+        observerRef.current.disconnect();
+        observerRef.current = null;
+      }
+    };
+  }, [containerRef, showSidePanel]);
```

Comment:
Does this work still if you remove `showSidePanel` from the `useEffect` dependencies?

Reply:
No it wasn't working without both dependencies